### DISCUSSION TITLE
Seeing continuous UE attach and detach to mme

### DISCUSSION THREAD

**Discussion by shalinjshah18** [magma/magma#7940] [Q&A]
Hi we are using srsUE and srsEnodeB to generate traffic to Magma AGW

But in AGW we continuously see Attach and detach of UE to mme.

See below logs

002958 Thu Jul 08 22:45:24 2021 7FA6B30EB700 INFO  S1AP   tasks/s1ap/s1ap_mme_nas_procedur:0873    Received Connection Establishment Confirm from MME_APP for ue_id = 9
002959 Thu Jul 08 22:45:24 2021 7FA6B30EB700 NOTIC S1AP   tasks/s1ap/s1ap_mme_nas_procedur:1123   [1] Send S1AP_INITIAL_CONTEXT_SETUP_REQUEST message MME_UE_S1AP_ID = 0x00000009 eNB_UE_S1AP_ID = 0x000030
002960 Thu Jul 08 22:45:24 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_bearer.c  :1544    Received MME_APP_INITIAL_CONTEXT_SETUP_RSP from S1AP for ue_id = (9)
002961 Thu Jul 08 22:45:24 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_bearer.c  :1472   [1010123456789] Sending S11 MODIFY BEARER REQ to SPGW for ue_id = (9), teid = (8)
002962 Thu Jul 08 22:45:24 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_bearer.c  :4427   [1010123456789] Sending S11 modify bearer req message to SPGW task for ue_id 0x00000009
002963 Thu Jul 08 22:45:24 2021 7FA6B60F1700 WARNI MME-AP tasks/mme_app/mme_app_context.c :0335   [56]  No IMSI hashtable for this IMSI
002964 Thu Jul 08 22:45:24 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_main.c    :0120    Received S11 MODIFY BEARER RESPONSE from SPGW

002965 Thu Jul 08 22:45:46 2021 7FA6B30EB700 INFO  S1AP   tasks/s1ap/s1ap_mme_handlers.c  :1131    UE CONTEXT RELEASE REQUEST with Cause_Type = Radio Network and Cause_Value = 20
002966 Thu Jul 08 22:45:46 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_itti_messa:0157   [1010123456789] Send Release Access Bearer Req for teid to spgw task 0x9
002967 Thu Jul 08 22:45:46 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_itti_messa:0086   [1010123456789] Sending UE Context Release Cmd to S1ap for (ue_id = 9)
UE Context Release Cause = (3)
002968 Thu Jul 08 22:45:46 2021 7FA6B48EE700 INFO  GTPv1- lib/openflow/controller/PagingAp:0116    Added paging flow rule for UE IP 192.168.128.18
002969 Thu Jul 08 22:45:46 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_context.c :1803    Received UE context release complete message for ue_id: 0x00000009
002970 Thu Jul 08 22:45:46 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_context.c :1383   [1010123456789] UE STATE - IDLE.

002971 Thu Jul 08 22:46:44 2021 7FA6B30EB700 INFO  S1AP   tasks/s1ap/s1ap_mme_handlers.c  :4583    Sent paging message over sctp for IMSI 001010123456789
002972 Thu Jul 08 22:46:44 2021 7FA6B30EB700 INFO  S1AP   tasks/s1ap/s1ap_mme_nas_procedur:0098    Received S1AP INITIAL_UE_MESSAGE ENB_UE_S1AP_ID 0x000031 assoc-id:12
002973 Thu Jul 08 22:46:44 2021 7FA6B30EB700 INFO  S1AP   tasks/s1ap/s1ap_mme_nas_procedur:0111    New Initial UE message received with eNB UE S1AP ID: 0x000031 assoc-id :12
002974 Thu Jul 08 22:46:44 2021 7FA6B30EB700 INFO  S1AP   tasks/s1ap/s1ap_mme_itti_messagi:0171    Sending Initial UE Message to MME_APP, enb_ue_s1ap_id : 0x000031
002975 Thu Jul 08 22:46:44 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_bearer.c  :0557    Received MME_APP_INITIAL_UE_MESSAGE from S1AP
002976 Thu Jul 08 22:46:44 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_bearer.c  :0702   [1010123456789] INITIAL_UE_MESSAGE RCVD
mme_ue_s1ap_id  = 9
enb_ue_s1ap_id  = 49
002977 Thu Jul 08 22:46:44 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_bearer.c  :0711   [1010123456789] Sending NAS Establishment Indication to NAS for ue_id = (9)
002978 Thu Jul 08 22:46:44 2021 7FA6B60F1700 INFO  NAS-EM tasks/nas/emm/sap/emm_as.c      :0174    EMMAS-SAP - Received primitive EMMAS_ESTABLISH_REQ (205)
002979 Thu Jul 08 22:46:44 2021 7FA6B60F1700 INFO  NAS-EM tasks/nas/emm/sap/emm_as.c      :0740    EMMAS-SAP - Received AS connection establish request
002980 Thu Jul 08 22:46:44 2021 7FA6B60F1700 INFO  NAS-EM tasks/nas/emm/sap/emm_as.c      :0879    EMMAS-SAP - Message Type = SERVICE_REQUEST(0x4d) for (ue_id = 9)
002981 Thu Jul 08 22:46:44 2021 7FA6B60F1700 INFO  NAS-EM tasks/nas/emm/sap/emm_recv.c    :0698    EMMAS-SAP - Received Service Request message for (ue_id = 9)
002982 Thu Jul 08 22:46:44 2021 7FA6B60F1700 INFO  NAS-EM tasks/nas/emm/sap/emm_as.c      :0174    EMMAS-SAP - Received primitive EMMAS_ESTABLISH_CNF (206)
002983 Thu Jul 08 22:46:44 2021 7FA6B60F1700 INFO  NAS-EM tasks/nas/emm/sap/emm_as.c      :2017    EMMAS-SAP - Send AS connection establish confirmation for (ue_id = 9)
002984 Thu Jul 08 22:46:44 2021 7FA6B60F1700 ERROR MME-AP tasks/mme_app/mme_app_bearer.c  :0428   [1010123456789] No NAS PDU found ue 0x00000009
002985 Thu Jul 08 22:46:44 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_context.c :1413   [1010123456789] UE STATE - CONNECTED.
002986 Thu Jul 08 22:46:44 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_bearer.c  :0535   [1010123456789] MME APP : Sent Initial context Setup Request and Started guard timer for UE id  9 timer_id :f3
002987 Thu Jul 08 22:46:44 2021 7FA6B30EB700 INFO  S1AP   tasks/s1ap/s1ap_mme_nas_procedur:0873    Received Connection Establishment Confirm from MME_APP for ue_id = 9
002988 Thu Jul 08 22:46:44 2021 7FA6B30EB700 NOTIC S1AP   tasks/s1ap/s1ap_mme_nas_procedur:1123   [1] Send S1AP_INITIAL_CONTEXT_SETUP_REQUEST message MME_UE_S1AP_ID = 0x00000009 eNB_UE_S1AP_ID = 0x000031
002989 Thu Jul 08 22:46:45 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_bearer.c  :1544    Received MME_APP_INITIAL_CONTEXT_SETUP_RSP from S1AP for ue_id = (9)
002990 Thu Jul 08 22:46:45 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_bearer.c  :1472   [1010123456789] Sending S11 MODIFY BEARER REQ to SPGW for ue_id = (9), teid = (8)
002991 Thu Jul 08 22:46:45 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_bearer.c  :4427   [1010123456789] Sending S11 modify bearer req message to SPGW task for ue_id 0x00000009
002992 Thu Jul 08 22:46:45 2021 7FA6B60F1700 WARNI MME-AP tasks/mme_app/mme_app_context.c :0335   [56]  No IMSI hashtable for this IMSI
002993 Thu Jul 08 22:46:45 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_main.c    :0120    Received S11 MODIFY BEARER RESPONSE from SPGW
002994 Thu Jul 08 22:47:07 2021 7FA6B30EB700 INFO  S1AP   tasks/s1ap/s1ap_mme_handlers.c  :1131    UE CONTEXT RELEASE REQUEST with Cause_Type = Radio Network and Cause_Value = 20
002995 Thu Jul 08 22:47:07 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_itti_messa:0157   [1010123456789] Send Release Access Bearer Req for teid to spgw task 0x9
002996 Thu Jul 08 22:47:07 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_itti_messa:0086   [1010123456789] Sending UE Context Release Cmd to S1ap for (ue_id = 9)
UE Context Release Cause = (3)
002997 Thu Jul 08 22:47:07 2021 7FA6B48EE700 INFO  GTPv1- lib/openflow/controller/PagingAp:0116    Added paging flow rule for UE IP 192.168.128.18
002998 Thu Jul 08 22:47:07 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_context.c :1803    Received UE context release complete message for ue_id: 0x00000009
002999 Thu Jul 08 22:47:07 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_context.c :1383   [1010123456789] UE STATE - IDLE.
003000 Thu Jul 08 22:48:04 2021 7FA6B30EB700 INFO  S1AP   tasks/s1ap/s1ap_mme_handlers.c  :4583    Sent paging message over sctp for IMSI 001010123456789
003001 Thu Jul 08 22:48:05 2021 7FA6B30EB700 INFO  S1AP   tasks/s1ap/s1ap_mme_nas_procedur:0098    Received S1AP INITIAL_UE_MESSAGE ENB_UE_S1AP_ID 0x000032 assoc-id:12
003002 Thu Jul 08 22:48:05 2021 7FA6B30EB700 INFO  S1AP   tasks/s1ap/s1ap_mme_nas_procedur:0111    New Initial UE message received with eNB UE S1AP ID: 0x000032 assoc-id :12
003003 Thu Jul 08 22:48:05 2021 7FA6B30EB700 INFO  S1AP   tasks/s1ap/s1ap_mme_itti_messagi:0171    Sending Initial UE Message to MME_APP, enb_ue_s1ap_id : 0x000032
003004 Thu Jul 08 22:48:05 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_bearer.c  :0557    Received MME_APP_INITIAL_UE_MESSAGE from S1AP
003005 Thu Jul 08 22:48:05 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_bearer.c  :0702   [1010123456789] INITIAL_UE_MESSAGE RCVD
mme_ue_s1ap_id  = 9
enb_ue_s1ap_id  = 50
003006 Thu Jul 08 22:48:05 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_bearer.c  :0711   [1010123456789] Sending NAS Establishment Indication to NAS for ue_id = (9)
003007 Thu Jul 08 22:48:05 2021 7FA6B60F1700 INFO  NAS-EM tasks/nas/emm/sap/emm_as.c      :0174    EMMAS-SAP - Received primitive EMMAS_ESTABLISH_REQ (205)
003008 Thu Jul 08 22:48:05 2021 7FA6B60F1700 INFO  NAS-EM tasks/nas/emm/sap/emm_as.c      :0740    EMMAS-SAP - Received AS connection establish request
003009 Thu Jul 08 22:48:05 2021 7FA6B60F1700 INFO  NAS-EM tasks/nas/emm/sap/emm_as.c      :0879    EMMAS-SAP - Message Type = SERVICE_REQUEST(0x4d) for (ue_id = 9)
003010 Thu Jul 08 22:48:05 2021 7FA6B60F1700 INFO  NAS-EM tasks/nas/emm/sap/emm_recv.c    :0698    EMMAS-SAP - Received Service Request message for (ue_id = 9)
003011 Thu Jul 08 22:48:05 2021 7FA6B60F1700 INFO  NAS-EM tasks/nas/emm/sap/emm_as.c      :0174    EMMAS-SAP - Received primitive EMMAS_ESTABLISH_CNF (206)
003012 Thu Jul 08 22:48:05 2021 7FA6B60F1700 INFO  NAS-EM tasks/nas/emm/sap/emm_as.c      :2017    EMMAS-SAP - Send AS connection establish confirmation for (ue_id = 9)
003013 Thu Jul 08 22:48:05 2021 7FA6B60F1700 ERROR MME-AP tasks/mme_app/mme_app_bearer.c  :0428   [1010123456789] No NAS PDU found ue 0x00000009
003014 Thu Jul 08 22:48:05 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_context.c :1413   [1010123456789] UE STATE - CONNECTED.
003015 Thu Jul 08 22:48:05 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_bearer.c  :0535   [1010123456789] MME APP : Sent Initial context Setup Request and Started guard timer for UE id  9 timer_id :f6
003016 Thu Jul 08 22:48:05 2021 7FA6B30EB700 INFO  S1AP   tasks/s1ap/s1ap_mme_nas_procedur:0873    Received Connection Establishment Confirm from MME_APP for ue_id = 9
003017 Thu Jul 08 22:48:05 2021 7FA6B30EB700 NOTIC S1AP   tasks/s1ap/s1ap_mme_nas_procedur:1123   [1] Send S1AP_INITIAL_CONTEXT_SETUP_REQUEST message MME_UE_S1AP_ID = 0x00000009 eNB_UE_S1AP_ID = 0x000032
003018 Thu Jul 08 22:48:05 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_bearer.c  :1544    Received MME_APP_INITIAL_CONTEXT_SETUP_RSP from S1AP for ue_id = (9)
003019 Thu Jul 08 22:48:05 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_bearer.c  :1472   [1010123456789] Sending S11 MODIFY BEARER REQ to SPGW for ue_id = (9), teid = (8)
003020 Thu Jul 08 22:48:05 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_bearer.c  :4427   [1010123456789] Sending S11 modify bearer req message to SPGW task for ue_id 0x00000009
003021 Thu Jul 08 22:48:05 2021 7FA6B60F1700 WARNI MME-AP tasks/mme_app/mme_app_context.c :0335   [56]  No IMSI hashtable for this IMSI
003022 Thu Jul 08 22:48:05 2021 7FA6B60F1700 INFO  MME-AP tasks/mme_app/mme_app_main.c    :0120    Received S11 MODIFY BEARER RESPONSE from SPGW

Need help to see if problem is on UE side or Magma AGW side to keep UE stable.

**Comment by pshelar** [comment]
@ulaskozat @ardzoht can you have a look.
